app/controllers/application_controller.rb

```diff
@@ -1,14 +1,43 @@
 class ApplicationController < ActionController::API
-  def current_user
-    @current_user || authenticate_token
+  include ActionController::HttpAuthentication::Token::ControllerMethods
+
+  attr_reader :current_user
+  before_action :authenticate
```

Comment:
Using the `:before_action :authenticate` method here; would imply that controllers that expose _public_ endpoints should have a `:skip_before_action` e.g. the `UsersController`